From 08d04475418b2ab590c6f7ebea39af746b312090 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Sat, 4 Dec 2021 14:25:14 +0900
Subject: [PATCH] fix build format error

---
 src/meta.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meta.c b/src/meta.c
index d92d0d3..b2a3511 100644
--- a/src/meta.c
+++ b/src/meta.c
@@ -408,7 +408,7 @@ static void date_type_print(const struct expr *expr, struct output_ctx *octx)
 	 * Do our own printing. The default print function will print in
 	 * nanoseconds, which is ugly.
 	 */
-	nft_print(octx, "%lu", tstamp);
+	nft_print(octx, "%llu", tstamp);
 }
 
 static time_t parse_iso_date(const char *sym)
-- 
2.17.1

