From 5446262b375592e0d46ecf958097b9b0b1a3cc46 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Thu, 2 Dec 2021 19:50:09 -0800
Subject: [PATCH] fix time_t musl 1.2.x

---
 sys-utils/unshare.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys-utils/unshare.c b/sys-utils/unshare.c
index e5627d3..59824df 100644
--- a/sys-utils/unshare.c
+++ b/sys-utils/unshare.c
@@ -214,7 +214,7 @@ static void settime(time_t offset, clockid_t clk_id)
 	char buf[sizeof(stringify_value(ULONG_MAX)) * 3];
 	int fd, len;
 
-	len = snprintf(buf, sizeof(buf), "%d %ld 0", clk_id, offset);
+	len = snprintf(buf, sizeof(buf), "%d %lld 0", clk_id, offset);
 
 	fd = open("/proc/self/timens_offsets", O_WRONLY);
 	if (fd < 0)
-- 
2.17.1

