From a406ba94262519afaff469887ed7557b4a5510bb Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Sat, 2 Apr 2022 09:13:12 +0900
Subject: [PATCH] igb:fix build with kernel 5.15

---
 src/igb_ethtool.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/igb_ethtool.c b/src/igb_ethtool.c
index a0881df0f..2a28d7efa 100644
--- a/src/igb_ethtool.c
+++ b/src/igb_ethtool.c
@@ -2353,9 +2353,13 @@ static int igb_phys_id(struct net_device *netdev, u32 data)
 	return 0;
 }
 #endif /* HAVE_ETHTOOL_SET_PHYS_ID */
-
-static int igb_set_coalesce(struct net_device *netdev,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0)
+	static int igb_set_coalesce(struct net_device *netdev,
+			    struct ethtool_coalesce *ec,struct kernel_ethtool_coalesce *,struct netlink_ext_ack *)
+#else
+	static int igb_set_coalesce(struct net_device *netdev,
 			    struct ethtool_coalesce *ec)
+#endif
 {
 	struct igb_adapter *adapter = netdev_priv(netdev);
 	int i;
@@ -2435,9 +2439,13 @@ static int igb_set_coalesce(struct net_device *netdev,
 
 	return 0;
 }
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0)
+static int igb_get_coalesce(struct net_device *netdev,
+			    struct ethtool_coalesce *ec,struct kernel_ethtool_coalesce *,struct netlink_ext_ack *)
+#else
 static int igb_get_coalesce(struct net_device *netdev,
 			    struct ethtool_coalesce *ec)
+#endif
 {
 	struct igb_adapter *adapter = netdev_priv(netdev);
 
-- 
2.17.1

