From 15353b48e88fd1ee1c6cc01cfb43075b490888af Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Wed, 1 Dec 2021 19:05:53 -0800
Subject: [PATCH] fix iptables for musl 1.2.x

---
 iptables/xshared.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iptables/xshared.c b/iptables/xshared.c
index 71f6899..0e5b8fb 100644
--- a/iptables/xshared.c
+++ b/iptables/xshared.c
@@ -283,7 +283,7 @@ static int xtables_lock(int wait, struct timeval *wait_interval)
 
 		if (++i % 10 == 0) {
 			fprintf(stderr, "Another app is currently holding the xtables lock; "
-				"still %lds %ldus time ahead to have a chance to grab the lock...\n",
+				"still %llds %lldus time ahead to have a chance to grab the lock...\n",
 				time_left.tv_sec, time_left.tv_usec);
 		}
 
-- 
2.17.1

