From b2ddfcd6ef8a1b73ff7a7bae42b8cb8e9b1d0d31 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Tue, 8 Feb 2022 02:05:06 -0500
Subject: [PATCH] 12

---
 e2fsck/unix.c | 2 +-
 e2fsck/util.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/e2fsck/unix.c b/e2fsck/unix.c
index f267bae2c..127bb2fff 100644
--- a/e2fsck/unix.c
+++ b/e2fsck/unix.c
@@ -1451,7 +1451,7 @@ int main (int argc, char *argv[])
 
 		fputs("<?xml version=\"1.0\" encoding=\"utf-8\"?>\n",
 		      ctx->problem_logf);
-		fprintf(ctx->problem_logf, "<problem_log time=\"%lu\">\n",
+		fprintf(ctx->problem_logf, "<problem_log time=\"%llu\">\n",
 			(unsigned long) ctx->now);
 		fprintf(ctx->problem_logf, "<invocation prog=\"%s\"",
 			argv[0]);
diff --git a/e2fsck/util.c b/e2fsck/util.c
index 3fe3c9888..b7fc43fe3 100644
--- a/e2fsck/util.c
+++ b/e2fsck/util.c
@@ -122,7 +122,7 @@ void *e2fsck_allocate_memory(e2fsck_t ctx, unsigned long size,
 	printf("Allocating %lu bytes for %s...\n", size, description);
 #endif
 	if (ext2fs_get_memzero(size, &ret)) {
-		sprintf(buf, "Can't allocate %lu bytes for %s\n",
+		sprintf(buf, "Can't allocate %llu bytes for %s\n",
 			size, description);
 		fatal_error(ctx, buf);
 	}
-- 
2.17.1

