From afed8db4050bf89c7c1b69c1df554e28d2148768 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Wed, 5 Jan 2022 22:05:45 -0500
Subject: [PATCH] default enbale anon_mount

---
 block.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block.c b/block.c
index 9f5cebf..80c838c 100644
--- a/block.c
+++ b/block.c
@@ -1678,7 +1678,7 @@ static int main_detect(int argc, char **argv)
 	cache_load(0);
 	printf("config 'global'\n");
 	printf("\toption\tanon_swap\t'0'\n");
-	printf("\toption\tanon_mount\t'0'\n");
+	printf("\toption\tanon_mount\t'1'\n");
 	printf("\toption\tauto_swap\t'1'\n");
 	printf("\toption\tauto_mount\t'1'\n");
 	printf("\toption\tdelay_root\t'5'\n");
-- 
2.17.1

