From 39669cb4e445f1a4162065985282ead5731f999c Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Sat, 2 Apr 2022 01:39:31 +0900
Subject: [PATCH] fix buile kernel 5.15

---
 file.c    |  9 ++++++++-
 ntfs_fs.h |  4 ++++
 xattr.c   | 11 +++++++++--
 3 files changed, 21 insertions(+), 3 deletions(-)

diff --git a/file.c b/file.c
index f839afaaf..a09a25cc9 100644
--- a/file.c
+++ b/file.c
@@ -947,8 +947,15 @@ static ssize_t ntfs_compress_write(struct kiocb *iocb, struct iov_iter *from)
 			size_t cp, tail = PAGE_SIZE - off;
 
 			page = pages[ip];
-			cp = iov_iter_copy_from_user_atomic(page, from, off,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0) 
+
+				cp = copy_page_from_iter_atomic(page, off,
+							    min(tail, bytes),from);
+#else
+				cp = iov_iter_copy_from_user_atomic(page, from, off,
 							    min(tail, bytes));
+#endif
+
 			flush_dcache_page(page);
 			iov_iter_advance(from, cp);
 			copied += cp;
diff --git a/ntfs_fs.h b/ntfs_fs.h
index 90913cda2..3abbe9e6b 100644
--- a/ntfs_fs.h
+++ b/ntfs_fs.h
@@ -799,7 +799,11 @@ int ntfs_cmp_names_cpu(const struct cpu_str *uni1, const struct le_str *uni2,
 
 /* globals from xattr.c */
 #ifdef CONFIG_NTFS3_FS_POSIX_ACL
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0)
+struct posix_acl *ntfs_get_acl(struct inode *inode, int type,bool);
+#else
 struct posix_acl *ntfs_get_acl(struct inode *inode, int type);
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0)
 int ntfs_set_acl(struct user_namespace *mnt_userns, struct inode *inode,
 #else
diff --git a/xattr.c b/xattr.c
index 67621167c..664778943 100644
--- a/xattr.c
+++ b/xattr.c
@@ -534,7 +534,11 @@ static struct posix_acl *ntfs_get_acl_ex(
  *
  * inode_operations::get_acl
  */
-struct posix_acl *ntfs_get_acl(struct inode *inode, int type)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0)
+	struct posix_acl *ntfs_get_acl(struct inode *inode, int type,bool)
+#else
+	struct posix_acl *ntfs_get_acl(struct inode *inode, int type)
+#endif
 {
 	/* TODO: init_user_ns? */
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0)
@@ -665,8 +669,11 @@ static int ntfs_xattr_get_acl(
 	if (!(inode->i_sb->s_flags & MS_POSIXACL))
 #endif
 		return -EOPNOTSUPP;
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0)
+	acl = ntfs_get_acl(inode, type,true);
+#else
 	acl = ntfs_get_acl(inode, type);
+#endif
 	if (IS_ERR(acl))
 		return PTR_ERR(acl);
 
-- 
2.17.1

